src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -2694,7 +2694,8 @@ public function has_self_closing_flag() {
 	public function is_tag_closer() {
 		return (
 			self::STATE_MATCHED_TAG === $this->parser_state &&
-			$this->is_closing_tag
+			$this->is_closing_tag &&
+			'BR' !== $this->get_tag()
```

Comment:
Why this one special case for a void element?

Reply:
a good call for an explanatory comment. `BR` is special in that a closing tag like `</br>` is always treated like an opening tag, and also as if it has no attributes.

in the Tag Processor we parse attributes but only store them if the tag is an opener. so if we report this here we should _only_ present this as an opening tag and the HTML API won't have to deal with the confusion of treating it as a closer.

at first I had this only in the HTML Processor, since it's a semantic issue and not a lexical one, but then I thought more about it and felt like there's never a reason to expose a BR closer, since they cannot exist.

Reply:
Noted in 677d03ca4b24efd92cd967838c1b9ffc9cfba634